src/pages/VisiumPerm.tsx

```diff
@@ -54,11 +55,29 @@ export default function VisiumPerm() {
       }
     });
   }, []);
-  const [current, send] = useMachine(formMachine);
 
+  const [current, send] = useMachine(formMachine);
+  const [warnBeforeSave, setWarnBeforeSave] = React.useState(false);
   const { serverError } = current.context;
 
   const onSubmit = async (values: RecordPermRequest) => {
+    const isLabwareStained = await stanCore
```

Comment:
only concern is if this request fails (which in theory it should never fail), users wont be alerted and the check wont work. Not sure this is a big deal though.